File: tests/fixtures/v2-http-sources/positive/request_sources.go
package fixture

import "net/http"

func review(r *http.Request) {
	body := r.FormValue("email")
	query := r.URL.Query().Get("q")
	path := r.PathValue("id")
	header := r.Header.Get("X-Trace")
	cookie, _ := r.Cookie("session")
	_, _, _, _, _ = body, query, path, header, cookie
}

File: tests/fixtures/v2-http-sources/positive/RequestSources.java
class RequestSources {
    void review(HttpServletRequest request) throws Exception {
        String query = request.getParameter("q");
        var body = request.getInputStream();
        String path = request.getPathInfo();
        String header = request.getHeader("X-Trace");
        var cookies = request.getCookies();
    }
}
